stdlib/string.gr

```diff
@@ -966,30 +978,24 @@ let _OFFSET_NAME = "offset"
 
 @unsafe
 let grainToWasmNumber = (num, name) => {
-  use WasmI32.{ (&), (>>), (<) }
-  let num = WasmI32.fromGrain(num)
-  if (WasmI32.eqz(num & 1n)) {
-    let str = " argument must be an integer"
-    throw Exception.InvalidArgument(name ++ str)
-  }
-  let num = num >> 1n
-  if (num < 0n) {
-    let str = " argument must be non-negative"
-    throw Exception.InvalidArgument(name ++ str)
+  use WasmI32.{ (<) }
+  let _num = convSimpleNumber(num, name ++ " argument must be an integer")
```

Comment:
A note is that in the nominal case this is now less efficient due to needing to do this string allocation + concat every time rather than only in the bad case